apollo-compiler/src/test/graphql/com/example/deprecated_merged_field/java/operationBased/deprecated_merged_field/adapter/CatQuery_ResponseAdapter.java.expected

```diff
@@ -77,9 +77,9 @@ public class CatQuery_ResponseAdapter {
       }
 
       Assertions.checkFieldNotMissing(__typename, "__typename");
-      reader.rewind();
       AnimalFragment animalFragment = null;
       if (BooleanExpressions.evaluate(new BooleanExpression.Element<BTerm>(new BPossibleTypes("Cat","Dog")), Collections.emptySet(), __typename)) {
+        reader.rewind();
```

Comment:
Because `endObject()` is now more strict, we need to make sure to rewind() only if really required